src/engine/BMGame.php

```diff
@@ -623,23 +526,20 @@ protected function load_swing_values_from_previous_round() {
     }
 
     protected function load_option_values_from_previous_round() {
-        if (!isset($this->optValueArrayArray)) {
-            return;
-        }
-
-        foreach ($this->optValueArrayArray as $playerIdx => $optionValueArray) {
-            if (!empty($optionValueArray)) {
-                $dieIndicesWithoutReserve = $this->die_indices_without_reserve($playerIdx);
+        foreach ($this->playerArray as $playerIdx => $player) {
+            if (empty($player->optValueArray)) {
+                continue;
+            }
 
-                foreach ($optionValueArray as $dieIdx => $optionValue) {
-                    $die = $this->playerArray[$playerIdx]
-                                ->activeDieArray[$dieIndicesWithoutReserve[$dieIdx]];
-                    if (!($die instanceof BMDieOption)) {
-                        throw new LogicException('Die must be an option die.');
-                    }
+            $dieIndicesWithoutReserve = $this->die_indices_without_reserve($playerIdx);
```

Comment:
Again, here, if `die_indices_without_reserve()` takes a mandatory player argument and acts just on the dice of that player, why not move it to `BMPlayer`?